src/lib/init.ts

```diff
@@ -312,92 +295,228 @@ type InitTSorJSProjectArg = {
+          }),
+        },
+        targetType: isJS ? "js" : "ts",
+        outputPath: frontendCompiledAquaPath,
+      });
+    })(),
+  ]);
+}
+
+function getIndexHTMLContent(isJS: boolean) {
+  return `<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" href="data:,">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Fluence</title>
+  </head>
+  <body>
+    <div id="app"></div>
+    <script type="module" src="/${relative(
+      getFrontendPath(),
+      getFrontendIndexTSorJSPath(isJS),
+    )}"></script>
+  </body>
+</html>
+`;
+}
 
-  const packageJson = {
-    type: "module",
-    version: "1.0.0",
-    main: indexFileName,
-    scripts: {
-      start: `node${isJS ? "" : "--loader ts-node/esm"} ${relative(
-        projectRootDir,
-        indexFilePath,
-      )}`,
-      ...(isJS ? {} : { build: "tsc -b" }),
+function getIndexJsContent(isJS: boolean) {
+  return `import { Fluence } from "@fluencelabs/js-client";
+import relays from "./relays.json" assert { type: "json" };
+import {
+  helloWorld,
+  helloWorldRemote,
+  runDeployedServices,
+  showSubnet,
+} from "./${relative(
+    getFrontendSrcPath(),
+    join(getFrontendCompiledAquaPath(), "main.js"),
+  )}";
+
+const appEl =
+  document.querySelector("#app") ??
+  (() => {
+    throw new Error("#app element is not found");
+  })();
+
+const buttons = [
+  {
+    fnName: "helloWorld",
+    fn() {
+      return helloWorld("Fluence");
     },
-    keywords: ["fluence"],
-    dependencies: {
-      [JS_CLIENT_NPM_DEPENDENCY]:
-        CLIPackageJSON.dependencies[JS_CLIENT_NPM_DEPENDENCY],
+  },
+  {
+    fnName: "helloWorldRemote",
+    fn() {
+      return helloWorldRemote("Fluence");
+    },
+  },
+  {
+    fnName: "showSubnet",
+    fn() {
+      return showSubnet();
     },
-    ...(isJS
-      ? {}
-      : {
-          devDependencies: {
-            "ts-node": CLIPackageJSON.devDependencies["ts-node"],
-            typescript: CLIPackageJSON.devDependencies["typescript"],
-          },
-        }),
-  } as const;
+  },
+  {
+    fnName: "runDeployedServices",
+    fn() {
+      return runDeployedServices();
+    },
+  },
+];
+
+(async () => {
+  try {
+    await Fluence.connect(relays[0].multiaddr, {});
+  } catch (error) {
+    appEl.innerHTML = stringifyError(error);
+    throw error;
+  }
 
-  await writeFile(
-    getPackageJSONPath(),
-    `${jsonStringify(packageJson)}\n`,
-    FS_OPTIONS,
-  );
+  appEl.innerHTML = buttons.map(({ fnName }) => button(fnName)).join("\\n");
+
+  requestAnimationFrame(() => {
+    buttons.forEach(({ fnName, fn }) => {
+      const buttonEl =
+        document.querySelector(\`#\${fnName}\`) ??
+        (() => {
+          throw new Error(\`Button with id="\${fnName}" is not found\`);
+        })();
+
+      buttonEl.addEventListener("click", () => {
+        runAquaFunction(fn);
+      });
+    });
+  });
+})();
```

Comment:
No need to use `requestAnimationFrame`  optimization here i think. You make very simple operation and just once. That function usually used for DOM animation which optimized to be updated only when your monitor showing next frame.

Reply:
restructured this part with createElement like in other places in the code so there is no need to find by id and stuff